app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt

```diff
@@ -33,11 +33,12 @@ class MessageListAdapter internal constructor(
         context: Context,
         theme: Resources.Theme,
         private val res: Resources,
-        private val fragment: MessageListFragment,
         private val layoutInflater: LayoutInflater,
         private val messageHelper: MessageHelper,
         private val contactsPictureLoader: ContactPictureLoader,
         private val accountRetriever: AccountRetriever,
+        private val toggleMessageSelectWithAdapterPosition: (Int) -> Unit,
```

Comment:
As you probably have noticed, using a function type like this makes it very ugly to use from Java. I think I'd prefer to have all callbacks in an interface. Something like `MessageListItemActionListener`.

Reply:
Done in a separate commit.